internal/handlers/pg/pgdb/pool.go

```diff
@@ -413,3 +440,97 @@ func (pgPool *Pool) SchemaStats(ctx context.Context, schema, collection string)
 	}
 	return &res, nil
 }
+
+// CreateSettingsTable creates FerretDB settings table.
+func (pgPool *Pool) CreateSettingsTable(ctx context.Context, db string) error {
+	sql := `CREATE TABLE ` + pgx.Identifier{db, collectionPrefix + "settings"}.Sanitize() + ` (_jsonb jsonb)`
+	_, err := pgPool.Exec(ctx, sql)
+	if err != nil {
+		return err
+	}
+
+	settings := must.NotFail(types.NewDocument("collections", must.NotFail(types.NewDocument())))
+	sql = fmt.Sprintf(`INSERT INTO %s (_jsonb) VALUES ($1)`, pgx.Identifier{db, collectionPrefix + "settings"}.Sanitize())
+	_, err = pgPool.Exec(ctx, sql, must.NotFail(fjson.Marshal(settings)))
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	return nil
+}
+
+func (pgPool *Pool) GetTableName(ctx context.Context, db, collection string) (string, error) {
+	tableExists, err := pgPool.TableExists(ctx, db, collectionPrefix+"settings")
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+	if !tableExists {
+		err = pgPool.CreateSettingsTable(ctx, db)
+		if err != nil {
+			return "", lazyerrors.Error(err)
+		}
+	}
+
+	sql := `SELECT _jsonb FROM ` + pgx.Identifier{db, collectionPrefix + "settings"}.Sanitize()
```

Comment:
So, in fact, in this table we have only one raw that stores all the settings for everything? Not like "one line - one collection"? 

I wonder how does it look like from the race condition perspective? Let's say we got a few parallel requests to update settings (e.g. create settings for a few different collections). How would it be handled?

Reply:
Yeah, we store a single row with a document that stores all the settings for a schema.

On a previous iteration, there would be a chance that we create several documents but right now all access to that table is going to be under the transaction, so I guess it should be ok.